digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -0,0 +1,1739 @@
+                case LOCAL: {
+                    if (file.template()) {
+                        try {
+                            String content = workspace.templateFile(templateEngine, reference.filename(), UTF_8, params);
+                            byte[] bytes = content.getBytes(UTF_8);
+                            ObjectMetadata metadata = new ObjectMetadata();
+                            metadata.setContentLength(bytes.length);
+                            return new PutObjectRequest(uri.getBucket(), uri.getKey(), new ByteArrayInputStream(bytes), metadata);
+                        }
+                        catch (IOException | TemplateException e) {
+                            throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                        }
+                    }
+                    else {
+                        return new PutObjectRequest(uri.getBucket(), uri.getKey(), workspace.getFile(reference.filename()));
+                    }
+                }
+                case RESOURCE: {
+                    byte[] bytes;
+                    try {
+                        bytes = Resources.toByteArray(new URL(reference.reference().get()));
+                    }
+                    catch (IOException e) {
+                        throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                    }
+                    ObjectMetadata metadata = new ObjectMetadata();
+                    metadata.setContentLength(bytes.length);
+                    return new PutObjectRequest(uri.getBucket(), uri.getKey(), new ByteArrayInputStream(bytes), metadata);
+                }
+                case DIRECT:
+                    byte[] bytes = reference.contents().get();
+                    ObjectMetadata metadata = new ObjectMetadata();
+                    metadata.setContentLength(bytes.length);
+                    return new PutObjectRequest(uri.getBucket(), uri.getKey(), new ByteArrayInputStream(bytes), metadata);
+                case S3:
+                default:
+                    throw new AssertionError();
+            }
+        }
+
+        void tryCleanup()
+        {
+            if (!staging.isPresent()) {
+                return;
+            }
+
+            String bucket = staging.get().getBucket();
+            ListObjectsRequest req = new ListObjectsRequest()
+                    .withBucketName(bucket)
+                    .withPrefix(staging.get().getKey());
+            do {
+                ObjectListing res = s3.listObjects(req);
+                String[] keys = res.getObjectSummaries().stream()
+                        .map(S3ObjectSummary::getKey)
+                        .toArray(String[]::new);
+                for (String key : keys) {
+                    logger.info("Removing s3://{}/{}", bucket, key);
+                }
+                s3.deleteObjects(new DeleteObjectsRequest(bucket).withKeys(keys));
```

Comment:
When `s3.deleteObjects()` throws an exception, we should continue to delete the rest of files? Hmm, but it might cause infinite loop... Maybe we'd better leave here as is.